aip/general/0193.md

```diff
@@ -210,75 +302,291 @@ map for the error payload to be backwards compatible, even if the value
+method to update an error message for a service is to add
+[`google.rpc.LocalizedMessage`](https://github.com/googleapis/googleapis/blob/e9897ed945336e2dc967b439ac7b4be6d2c62640/google/rpc/error_details.proto#L275-L285)
+to
+[`Status.details`](https://github.com/googleapis/googleapis/blob/aeae5ea2b01ece6c0cee046ae84b881cdc62b95d/google/rpc/status.proto#L46-L48).
+`LocalizedMessage` is meant for displaying messages to end users.
+
+The error message presented in `LocalizedMessage.message` **should** be
+the same as `Status.message` or it **should** be an alternate message.
+Add as much information as the consumer of the error needs to resolve
+the error, but succinctly.
+
+Reasons to present the same error message in both locations include the
+following:
+
+- The service plans to localize either immediately or in the near future. See,
+   "[Localization](https://google.aip.dev/193#localization)".
+- This practice enables clients to find an error message consistently in
+  one location, `LocalizedMessaage.message`, across all methods of the
+  API Service.
+
+Reasons to present an error message in `LocalizedMessage.message` that
+differs from `Status.message` include the following:
+
+- The service requires an end-user-facing error message that differs from the
+  "debug message".
+- Ongoing, iterative error message improvements are expected.
+
+When including `LocalizedMessage`, both `locale` and `message` **must**
+be populated. If the service is to be localized, the value of `locale`
+**must** change dynamically. See
+"[Localization](https://google.aip.dev/193#localization)".  Otherwise,
+`locale` **must** always present the service's default locale; for
+example, "en-US".
+
+When adding an error message using `LocalizedMessage`, `ErrorInfo`
+**must** be introduced either before or at the same time. If there are
+dynamic segments found in the text, the values of these variables
+**must** be included in `ErrorInfo.metadata`. See, "[Dynamic
+variables](#dynamic-variables)".
+
+**Warning:** If `LocalizedMessage` is released without `ErrorInfo`, the
+same [concerns](#status-message-warning) regarding client misuse of
+textual error messages apply.
+
+#### Message Layout
+
+Actionable error messages are consumed by human actors, often brokered
+by a specific surface. For these messages to be maximally useful and
+actionable, an effective layout is imperative.
+
+The following layout guidelines make error messages easy to read or
+scan.
+
+The presentation depends on the product surface, so these guidelines
+explain best practices for a standard layout across common product
+surfaces such as gcloud, Pantheon, or Terraform.
+
+- Place the Error description in the first line (required)
+- List each instance of metadata in a new line, indented
+- Place the Resolution in a newline after a line space
+- Include a hyperlink for documentation as part of the description
+
+The following example shows an actionable error message for the gcloud
```

Comment:
this content should be moved to Cloud SDK AIPs (https://google.aip.dev/cloud#sdk), but I can do that in a follow-up CL to get this unblocked.

Reply:
for now I suggest removing this paragraph from here to keep this AIP generic.

Reply:
Do we want to remove the entire Message Layout section or just the bullet and example?